subsys/bluetooth/controller/hal/swi.h

```diff
@@ -0,0 +1,7 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "hal/swi_vendor_hal.h"
```

Comment:
@ioannisg : Question: Why are you adding this header?
(this new swi.h would seem to be internal to the LLL, so no need to expose it to the rest of the controller, and of creating this extra headers, or?)

Reply:
@aescolar I submitted the change, I was trying to follow the pattern here (https://github.com/zephyrproject-rtos/zephyr/tree/master/subsys/bluetooth/controller/hal) in line with radio.h and ticker.h

If this is ok, maybe we remove others like ccm and ecb (and the now added swi.h) specific to nordic as future work in a new Issue/PR.

Reply:
But the ccm and ecb HAL APIs are used by the ULL, or?
is only this new header that is just for the LLL?

Reply:
The top level hal folder is there partly for co-existence of legacy and split controller, and then there is the necessity of ULL referencing definitions that are vendor defined (ex., HAL_TICKER_US_TO_TICKS etc.).

The reason I added swi.h in the top folder was to be able to use definitions of IRQ numbers in legacy controller, but then I have not updated legacy controller with the abstractions (not a priority to support it any more).

Reply:
Agree with @cvinayak that the main reason, here, was compliance. 
ll_sw does not contain, itself, a "hal" directory, so we add this header. We could have this internally in the nordic/ sub-directory, but this is specific to SPLIT architecture, and does not allow to extend this "abstraction" to the legacy controller.

It is a style thing, though; it aligns with radio.h and ticker.h which _similarly_ abstract the vendor-specific hal inclusions.

Reply:
Up to you guys. It is just that other LLL need to provide the other headers, while they will not need to provide this new swi.h